dask/base.py

```diff
@@ -96,7 +97,7 @@ def compute(self, **kwargs):
     @classmethod
     def _get(cls, dsk, keys, get=None, **kwargs):
         get = get or _globals['get'] or cls._default_get
-        dsk2 = cls._optimize(dsk, keys, **kwargs)
+        dsk2 = cls._optimize(dict(dsk), keys, **kwargs)
```

Comment:
What is the performance of `dict(ShareDict)`? Perhaps a dedicated `ensure_dict` function could be faster? (and also avoid creating a new dict if `dsk` is already a dict...).